packages/mdc-menu-surface/foundation.js

```diff
@@ -286,12 +306,22 @@ class MDCMenuSurfaceFoundation extends MDCFoundation {
    */
   getHorizontalOriginOffset_(corner) {
     const {anchorWidth} = this.measures_;
+    // isRightAligned corresponds to using the 'right' property on the surface.
     const isRightAligned = Boolean(corner & MenuSurfaceCornerBit.RIGHT);
     const avoidHorizontalOverlap = Boolean(this.anchorCorner_ & MenuSurfaceCornerBit.RIGHT);
+
     if (isRightAligned) {
-      const rightOffset = avoidHorizontalOverlap ? anchorWidth - this.anchorMargin_.left : this.anchorMargin_.right;
+      let rightOffset = avoidHorizontalOverlap ? anchorWidth - this.anchorMargin_.left : this.anchorMargin_.right;
+
+      // Hoisted elements positioning doesn't account for the scrollbar, so the right property needs to be reduced by
+      // the difference between the window and body width.
+      if (this.hoistedElement_) {
```

Comment:
Moving this above the `let rightOffset` line will allow you to do an early return:
```
if (this.hoistedElement_) {
return ...
}
return avoidHorizontalOverlap ? anchorWidth - this.anchorMargin_.left : this.anchorMargin_.right;
```

Reply:
The nested line uses the `rightOffset` variable.

Reply:
ohh you're so right :+1: